### DISCUSSION TITLE
Error running an example in PorousFlow modulus: Need Exodus reader to restart variables

### DISCUSSION THREAD

**Discussion by sinaheidari22** [idaholab/moose#21367] [Q&A Modules: General] [answered]
Dear all, 

I was trying to run an available example in the PorousFlow module considering a two-part simulation which is restarting a simulation using previous simulations.     Here is the file: (modules/porous_flow/examples/restart/gas_injection.i)

I got this error 
*** ERROR ***
The following error occurred in the object "MOOSE Problem", of type "FEProblem".

Need Exodus reader to restart variables but the reader is not available
Use either FileMesh with an Exodus mesh file or FileMeshGenerator with an Exodus mesh file and with use_for_exodus_restart equal to true.

Could you please advise on that, 
Thanks for reading.

**Comment by GiudGiud** [comment, marked as answer]
Hello

You need to modify your Mesh block to
```
[Mesh]
  uniform_refine = 1
  [file]
    type = FileMeshGenerator
    file = gravityeq_out.e
    use_for_exodus_restart = true
  []
  [translate]
    type = TransformGenerator
    transform = TRANSLATE
    vector_value = '0.1 0 0'
    input = file
  []
[]
```

**Reply by sinaheidari22** [reply]
Thank you for your prompt response. 

The issue is solved now.
